core/internal/processlib/processlib_linux.go

```diff
@@ -0,0 +1,24 @@
+package processlib
+
+import (
+	"os"
+	"syscall"
+)
+
+const (
+	PRCTL_SYSCALL    = 157
+	PR_SET_PDEATHSIG = 1
+)
+
+func ShutdownOnParentExit(parentPid int) bool {
+	_, _, errno := syscall.Syscall(
+		uintptr(PRCTL_SYSCALL), uintptr(PR_SET_PDEATHSIG), uintptr(syscall.SIGKILL), 0)
+	if errno != 0 {
+		os.Exit(127 + int(errno))
+	}
+	// Check again because the process could have exited right before the syscall was made
```

Comment:
also say that in that case os.Getppid() would return 1 in case they don't know

Reply:
Yep.. which would mean that it wouldnt match the parent, assuming we didnt start wandb_core from process=1

Reply:
same as above let's explain why error code 1?

Reply:
have to do something no zero... it is arbitrary. prior reviewer didnt like 2, so i chose 1 :)
we do not check this anywhere, and the parent process has already gone away, so it really feels arbitrary.
i added a comment